src/repositories/user/UserRepository.ts

```diff
@@ -1,26 +1,30 @@
 import * as mongoose from 'mongoose';
 import { userModel } from './UserModel';
 import IUserModel from './IUserModel';
+import VersionableRepository from '../versionable/VersionableRepository';
+export default class UserRepository extends VersionableRepository<IUserModel, mongoose.Model<IUserModel>> {
 
-export default class UserRepository {
     public static generateObjectId() {
         return String(mongoose.Types.ObjectId());
     }
-    public static findOne(query): mongoose.DocumentQuery<IUserModel, IUserModel, {}> {
-        return userModel.findOne(query).lean();
+    constructor() {
+        super(userModel);
     }
-    public find(query, projection ?: any, options ?: any): any {
-        return userModel.find(query, projection, options);
+    public static readOne(query): mongoose.DocumentQuery<IUserModel, IUserModel, {}> {
+        return userModel.findOne(query).lean();
     }
+
     public create(data: any): Promise<IUserModel> {
-        console.log('UserRepository:: create', data);
+        console.log('User Data:', data);
         const id = UserRepository.generateObjectId();
         const model = new userModel({
             _id: id,
             ...data,
+            originalId: id,
         });
         return model.save();
     }
+
```

Comment:
No need to create user for here instead use versionable create

Reply:
changes done